src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/CollectionPropertiesShouldBeReadOnly.cs

```diff
@@ -135,5 +135,25 @@ private static bool Inherits(ITypeSymbol symbol, ITypeSymbol baseType)
             Debug.Assert(baseType.Equals(baseType.OriginalDefinition));
             return symbol?.OriginalDefinition.Inherits(baseType) ?? false;
         }
+
+        private static ImmutableHashSet<INamedTypeSymbol> GetIImmutableInterfaces(Compilation compilation)
```

Comment:
This is used in only one place so I decided to move the code here. If you prefer, I can add it to the provider.